src/main/java/org/jabref/gui/preferences/websearch/WebSearchTabViewModel.java

```diff
@@ -138,7 +139,11 @@ public void setValues() {
         grobidEnabledProperty.setValue(grobidPreferences.isGrobidEnabled());
         grobidURLProperty.setValue(grobidPreferences.getGrobidURL());
 
-        apiKeys.setValue(FXCollections.observableArrayList(preferences.getImporterPreferences().getApiKeys()));
+        // Initialize apiKeys with a deep copy of the actual API Keys
+        apiKeys.setAll(preferences.getImporterPreferences().getApiKeys().stream()
+                                  .map(apiKey -> new FetcherApiKey(apiKey.getName(), apiKey.shouldUse(), apiKey.getKey()))
+                                  .collect(Collectors.toList()));
```

Comment:
Can be simplified 
```suggestion
                                  .toList());
```

Reply:
Careful, Collectors.toList() creates a new mutable array lust, .toList() an immutable one.

Reply:
OKay, so I reckon we'll continue using .`collect(Collectors.toList())` to obtain a mutable list, since apiKeys needs to be mutable to allow modifications in the UI ? 

I am a bit unsure about it.

Reply:
Yes, we need an immutable list here.